internal/descriptor/grpc_api_configuration.go

```diff
@@ -6,23 +6,13 @@ import (
 	"strings"
 
 	"github.com/grpc-ecosystem/grpc-gateway/v2/internal/descriptor/apiconfig"
-	"google.golang.org/protobuf/encoding/protojson"
-	"sigs.k8s.io/yaml"
+	"gopkg.in/yaml.v3"
 )
 
 func loadGrpcAPIServiceFromYAML(yamlFileContents []byte, yamlSourceLogName string) (*apiconfig.GrpcAPIService, error) {
-	jsonContents, err := yaml.YAMLToJSON(yamlFileContents)
+	var serviceConfiguration apiconfig.GrpcAPIService
+	err := yaml.Unmarshal(yamlFileContents, &serviceConfiguration)
 	if err != nil {
-		return nil, fmt.Errorf("failed to convert gRPC API Configuration from YAML in '%v' to JSON: %v", yamlSourceLogName, err)
-	}
-
-	// As our GrpcAPIService is incomplete, accept unknown fields.
-	unmarshaler := protojson.UnmarshalOptions{
-		DiscardUnknown: true,
-	}
```

Comment:
I think this was the reason we were using the other YAML parser, so we definitely still need this behavior. Can you make a test that checks that we can still unmarshal unknown fields?

Reply:
You were right on the gold! Totally missed the protojson Marshalling
Should be happy now